xwiki-commons-core/xwiki-commons-configuration/xwiki-commons-configuration-default/src/test/java/org/xwiki/configuration/internal/SystemEnvConfigurationSourceTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.configuration.internal;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.xwiki.properties.ConverterManager;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+/**
+ * Validate {@link SystemEnvConfigurationSource}.
+ * 
+ * @version $Id$
+ */
+@ComponentTest
+class SystemEnvConfigurationSourceTest
+{
+    public static class TestSystemEnvConfigurationSource extends SystemEnvConfigurationSource
+    {
+        private final Map<String, String> env = new LinkedHashMap<>();
+
+        @Override
+        Map<String, String> getenv()
+        {
+            return env;
+        }
+
+        @Override
+        String getenv(String name)
+        {
+            return env.get(name);
+        }
+    }
+
+    @InjectMockComponents
+    private TestSystemEnvConfigurationSource configuration;
+
+    @MockComponent
+    private ConverterManager converterManager;
+
+    @Test
+    void getProperty()
+    {
+        assertNull(this.configuration.getProperty(null));
+        assertNull(this.configuration.getProperty("doesnotexist"));
+        assertNull(this.configuration.getProperty("key", Integer.class));
+        assertEquals(1, this.configuration.getProperty("key", 1));
+
+        this.configuration.getenv().put("key", "value");
```

Comment:
hmmm the package private was for the tests?

Reply:
Yes, it was.